### PR TITLE
Test Feature

### PR DESCRIPTION
**Pull request by dheerajsachdeva**
## Testing in React

- Set up React Testing Library.
- Unit tests for the files operate.js and calculate.js using Jest.
- Unit tests for all React components
- Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.
- Run full tests suite using Jest. All tests  pass.

### CODE REVIEW

**Review by osamaashraf6 — COMMENTED** [review]
Hi @dheerajsachdeva 😊, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🆘💀

- You have submitted a project, that no partner working it, so please try to let your partner commit to the project as This milestone is an [pair-programming] activity.😊💯

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/tests/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import renderer from 'react-test-renderer';
+import Calculator from '../components/Calculator';
+
+it('renders correctly', () => {
+  const tree = renderer
+    .create(<Calculator />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (DammyShittu) [inline src/tests/Calculator.test.js]: - [ ] Kindly add tests that simulate user interaction events like, for example in this component, performing a calculation. You can do this using the react testing library you installed. You visit this link for an idea of how to go about that: [https://www.robinwieruch.de/react-testing-library/](https://www.robinwieruch.de/react-testing-library/)

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @dheerajsachdeva and @Kevin-Mena,

## Highlights
✔️ Snapshot tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Calculator.test.js (hunk 1 above)
Comment (dheerajsachdeva) [reply src/tests/Calculator.test.js]: Done. Thanks

**Review by AnselemOdims — APPROVED** [review]
### Hi  @dheerajsachdeva 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**